db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,145 @@
+                                               else
+                                                 match_language(key).to_s
+                                               end
+      end
+    end
+
+    existing_primary_languages = Patient.where(purged: false).where.not(primary_language: nil).distinct.pluck(:primary_language)
+    existing_secondary_languages = Patient.where(purged: false).where.not(secondary_language: nil).distinct.pluck(:secondary_language)
+    # Create an translation in automatic_translations, unless we already have it defined in `custom_translations`
+    (existing_primary_languages | existing_secondary_languages).each do |el|
+      automatic_translations[el.to_sym] = (match_language(el).to_s || nil) unless custom_translations.key?(el.to_sym)
+    end
+    ActiveRecord::Base.transaction do
+      automatic_translations.each do |key, value|
+        # Patient records updated here are solely updated (no History Items included)
+        Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+        Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+      end
+      custom_translations.each do |key, value|
+        key = key.to_s
+        if value === 'bilingual english/spanish'
+          Patient.where(purged: false).where(primary_language: key).update_all({ primary_language: 'eng', secondary_language: 'spa' })
+        elsif value === 'bilingual spanish/english'
+          Patient.where(purged: false).where(primary_language: key).update_all({ primary_language: 'spa', secondary_language: 'eng' })
+        elsif TRANSLATION_COMMENTS.include?(key)
+          note = "Primary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s primary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id,created_by,comment,history_type,created_at,updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND primary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+
+          note = "Secondary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s secondary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id,created_by,comment,history_type,created_at,updated_at)
```

Comment:
super nitpicky, but for styling purposes, can we add a space in between these fields

Reply:
Nitpicky is good! Fixed.